expression/builtin_time_vec.go

```diff
@@ -1732,11 +1732,72 @@ func (b *builtinToDaysSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinDateFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDateFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	dateBuf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dateBuf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, dateBuf); err != nil {
+		return err
+	}
+	times := dateBuf.Times()
+
+	formatBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(formatBuf)
+	if err := b.args[1].VecEvalString(b.ctx, input, formatBuf); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+
+	for i := range times {
+		t := times[i]
+		if dateBuf.IsNull(i) || formatBuf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		formatMask := formatBuf.GetString(i)
+		// MySQL compatibility, #11203
+		// If format mask is 0 then return 0 without warnings
+		if formatMask == "0" {
+			result.AppendString("0")
+			continue
+		}
+
+		if t.InvalidZero() {
+			// MySQL compatibility, #11203
+			// 0 | 0.0 should be converted to null without warnings
+			n, err := t.ToNumber().ToInt()
+			isOriginalIntOrDecimalZero := err == nil && n == 0
+			// Args like "0000-00-00", "0000-00-00 00:00:00" set Fsp to 6
+			isOriginalStringZero := t.Fsp() > 0
+
+			result.AppendNull()
+
+			if isOriginalIntOrDecimalZero && !isOriginalStringZero {
+				continue
+			}
```

Comment:
`result.AppendNull()` should be in this branch?

Reply:
vector based DATE_FORMAT will exit when encountered first `handleInvalidTimeError`, is this the same behaviour with row based DATE_FORMAT?